### ISSUE TITLE
[1.14.4] Optifine causes all game text to dissapear/break

### ISSUE THREAD

**Issue by facetorched** [BluSunrize/ImmersiveEngineering#3857]
Loading optifine (and its 2 dependencies) with Immersive Engineering works completely fine but as soon as you open a world and hover over or hold an IE item, every instance of text in the game breaks. The game continues to be completely playable but I can't read anything until I restart the game.

![2020-05-13_17 26 31](https://user-images.githubusercontent.com/60113330/81872521-7fedf180-953f-11ea-996c-5c12330f79d8.png)
![broken text](https://user-images.githubusercontent.com/60113330/81872571-95fbb200-953f-11ea-9b87-cd1553d00d9f.JPG)

No crash log but here is latest.txt. All I did was  log on, open a world, type in chat before the issue has occurred then type something something after it has occurred https://pastebin.com/kmen4nB4

Modlist:
forge-1.14.4-28.2.11
ImmersiveEngineering-0.14-95
OptiFine_1.14.4_HD_U_F5
OptiForge-MC1.14.4-0.1.1
MixinBootstrap-1.0.3

**Comment by malte0811** [comment]
I'll look into this later today. It's a known issue, but until now I thought that no one had made a utility for running Optifine in a development environment in 1.14+ yet. But apparently OptiForge can do that, so I'll see if I can find anything.

**Comment by snitxmhm** [comment]
Hello,

> I'll look into this later today. It's a known issue, but until now I thought that no one had made a utility for running Optifine in a development environment in 1.14+ yet. But apparently OptiForge can do that, so I'll see if I can find anything.

Hello,malte0811, I met the same problem before. And when i saw the you fixed the issue, I cloned the source code and tried to complile on my own.
`./gradlew build --warning-mode all`
After successful compilation, it can be successfully loaded by the game. And the previous problem is solved.
But ther goes another issue: when I pull down the Immersive Engineering item menu to a certain level in creative mode, the game crashed down whit the errors:empty quad list!
```text
---- Minecraft Crash Report ----
// I just don't know what went wrong :(

Time: 5/19/20 6:25 PM
Description: Rendering item

java.lang.RuntimeException: Empty quad list!
	at blusunrize.immersiveengineering.client.models.ModelCoresample.getQuads(ModelCoresample.java:214) ~[?:?] {re:classloading}
	at blusunrize.immersiveengineering.client.models.BakedIEModel.func_200117_a(BakedIEModel.java:32) ~[?:?] {re:classloading}
	at net.minecraftforge.client.ForgeHooksClient.renderLitItem(ForgeHooksClient.java:637) ~[?:?] {re:mixin,re:classloading}
	at sun.reflect.GeneratedMethodAccessor49.invoke(Unknown Source) ~[?:?] {}
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_242] {}
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_242] {}
	at net.optifine.reflect.Reflector.call(Reflector.java:905) ~[?:?] {re:classloading}
	at net.optifine.reflect.ReflectorMethod.call(ReflectorMethod.java:145) ~[?:?] {re:mixin,re:classloading}
	at net.minecraft.client.renderer.ItemRenderer.func_191967_a(ItemRenderer.java:116) ~[?:?] {re:classloading,pl:accesstransformer:B,xf:OptiFine:default,xf:fml:durability101:durability101}
	at net.minecraft.client.renderer.ItemRenderer.func_191961_a(ItemRenderer.java:100) ~[?:?] {re:classloading,pl:accesstransformer:B,xf:OptiFine:default,xf:fml:durability101:durability101}
	at net.minecraft.client.renderer.ItemRenderer.func_180454_a(ItemRenderer.java:190) ~[?:?] {re:classloading,pl:accesstransformer:B,xf:OptiFine:default,xf:fml:durability101:durability101}
	at net.minecraft.client.renderer.ItemRenderer.func_191962_a(ItemRenderer.java:511) ~[?:?] {re:classloading,pl:accesstransformer:B,xf:OptiFine:default,xf:fml:durability101:durability101}
	at net.minecraft.client.renderer.ItemRenderer.func_184391_a(ItemRenderer.java:553) ~[?:?] {re:classloading,pl:accesstransformer:B,xf:OptiFine:default,xf:fml:durability101:durability101}
	at net.minecraft.client.gui.screen.inventory.ContainerScreen.func_146977_a(ContainerScreen.java:232) ~[?:?] {re:classloading,pl:runtimedistcleaner:A}
	at net.minecraft.client.gui.screen.inventory.ContainerScreen.render(ContainerScreen.java:96) ~[?:?] {re:classloading,pl:runtimedistcleaner:A}
	at net.minecraft.client.gui.DisplayEffectsScreen.render(DisplayEffectsScreen.java:45) ~[?:?] {re:classloading,pl:runtimedistcleaner:A}
	at net.minecraft.client.gui.screen.inventory.CreativeScreen.render(CreativeScreen.java:600) ~[?:?] {re:classloading,pl:runtimedistcleaner:A}
	at net.minecraftforge.client.ForgeHooksClient.drawScreen(ForgeHooksClient.java:354) ~[?:?] {re:mixin,re:classloading}
	at sun.reflect.GeneratedMethodAccessor19.invoke(Unknown Source) ~[?:?] {}
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_242] {}
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_242] {}
	at net.optifine.reflect.Reflector.callVoid(Reflector.java:766) ~[?:?] {re:classloading}
	at net.minecraft.client.renderer.GameRenderer.func_195458_a(GameRenderer.java:896) ~[?:?] {re:classloading,pl:accesstransformer:B,xf:OptiFine:default}
	at net.minecraft.client.Minecraft.func_195542_b(Minecraft.java:890) ~[?:?] {re:classloading,pl:accesstransformer:B,xf:fml:randompatches:RandomPatches Minecraft Transformer,pl:runtimedistcleaner:A,re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A}
	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:384) ~[?:?] {re:classloading,pl:accesstransformer:B,xf:fml:randompatches:RandomPatches Minecraft Transformer,pl:runtimedistcleaner:A,re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A}
	at net.minecraft.client.main.Main.main(SourceFile:155) ~[1.14.4-forge-28.2.16.jar:?] {re:classloading}
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_242] {}
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_242] {}
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_242] {}
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_242] {}
	at net.minecraftforge.fml.loading.FMLClientLaunchProvider.lambda$launchService$0(FMLClientLaunchProvider.java:56) ~[forge-1.14.4-28.2.16.jar:28.2] {}
	at cpw.mods.modlauncher.LaunchServiceHandlerDecorator.launch(LaunchServiceHandlerDecorator.java:37) [modlauncher-4.1.0.jar:?] {}
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:54) [modlauncher-4.1.0.jar:?] {}
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:72) [modlauncher-4.1.0.jar:?] {}
	at cpw.mods.modlauncher.Launcher.run(Launcher.java:81) [modlauncher-4.1.0.jar:?] {re:classloading}
	at cpw.mods.modlauncher.Launcher.main(Launcher.java:65) [modlauncher-4.1.0.jar:?] {re:classloading}


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Client thread
Stacktrace:
	at blusunrize.immersiveengineering.client.models.ModelCoresample.getQuads(ModelCoresample.java:214)
	at blusunrize.immersiveengineering.client.models.BakedIEModel.func_200117_a(BakedIEModel.java:32)
	at net.minecraftforge.client.ForgeHooksClient.renderLitItem(ForgeHooksClient.java:637)
	at sun.reflect.GeneratedMethodAccessor49.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at net.optifine.reflect.Reflector.call(Reflector.java:905)
	at net.optifine.reflect.ReflectorMethod.call(ReflectorMethod.java:145)
	at net.minecraft.client.renderer.ItemRenderer.func_191967_a(ItemRenderer.java:116)
	at net.minecraft.client.renderer.ItemRenderer.func_191961_a(ItemRenderer.java:100)
	at net.minecraft.client.renderer.ItemRenderer.func_180454_a(ItemRenderer.java:190)
	at net.minecraft.client.renderer.ItemRenderer.func_191962_a(ItemRenderer.java:511)

-- Item being rendered --
Details:
	Item Type: coresample
	Registry Name: immersiveengineering:coresample
	Item Damage: 0
	Item NBT: null
	Item Foil: false
Stacktrace:
	at net.minecraft.client.renderer.ItemRenderer.func_184391_a(ItemRenderer.java:553)
	at net.minecraft.client.gui.screen.inventory.ContainerScreen.func_146977_a(ContainerScreen.java:232)
	at net.minecraft.client.gui.screen.inventory.ContainerScreen.render(ContainerScreen.java:96)
	at net.minecraft.client.gui.DisplayEffectsScreen.render(DisplayEffectsScreen.java:45)
	at net.minecraft.client.gui.screen.inventory.CreativeScreen.render(CreativeScreen.java:600)
	at net.minecraftforge.client.ForgeHooksClient.drawScreen(ForgeHooksClient.java:354)
	at sun.reflect.GeneratedMethodAccessor19.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at net.optifine.reflect.Reflector.callVoid(Reflector.java:766)

-- Screen render details --
Details:
	Screen name: net.minecraft.client.gui.screen.inventory.CreativeScreen
	Mouse location: Scaled: (320, 146). Absolute: (1280.000000, 583.000000)
	Screen size: Scaled: (473, 247). Absolute: (1890, 986). Scale factor of 4.000000

-- Affected level --
Details:
	All players: 1 total; [ClientPlayerEntity['Einsam_'/203, l='MpServer', x=101.59, y=77.12, z=-38.00]]
	Chunk stats: Client Chunk Cache: 529, 342
	Level dimension: DimensionType{minecraft:overworld}
	Level name: MpServer
	Level seed: 0
	Level generator: ID 00 - default, ver 1. Features enabled: false
	Level generator options: {}
	Level spawn location: World: (128,66,-32), Chunk: (at 0,4,0 in 8,-2; contains blocks 128,0,-32 to 143,255,-17), Region: (0,-1; contains chunks 0,-32 to 31,-1, blocks 0,0,-512 to 511,255,-1)
	Level time: 11079 game time, 11079 day time
	Level storage version: 0x00000 - Unknown?
	Level weather: Rain time: 0 (now: false), thunder time: 0 (now: false)
	Level game mode: Game mode: creative (ID 1). Hardcore: false. Cheats: false
	Server brand: forge
	Server type: Integrated singleplayer server
Stacktrace:
	at net.minecraft.client.world.ClientWorld.func_72914_a(ClientWorld.java:574)
	at net.minecraft.client.Minecraft.func_71396_d(Minecraft.java:1750)
	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:400)
	at net.minecraft.client.main.Main.main(SourceFile:155)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at net.minecraftforge.fml.loading.FMLClientLaunchProvider.lambda$launchService$0(FMLClientLaunchProvider.java:56)
	at cpw.mods.modlauncher.LaunchServiceHandlerDecorator.launch(LaunchServiceHandlerDecorator.java:37)
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:54)
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:72)
	at cpw.mods.modlauncher.Launcher.run(Launcher.java:81)
	at cpw.mods.modlauncher.Launcher.main(Launcher.java:65)

-- System Details --
Details:
	Minecraft Version: 1.14.4
	Minecraft Version ID: 1.14.4
	Operating System: Linux (amd64) version 5.4.39-1-MANJARO
	Java Version: 1.8.0_242, Oracle Corporation
	Java VM Version: OpenJDK 64-Bit Server VM (mixed mode), Oracle Corporation
	Memory: 870521976 bytes (830 MB) / 2147483648 bytes (2048 MB) up to 2147483648 bytes (2048 MB)
	CPUs: 8
	JVM Flags: 8 total; -Xss1M -Xmx2G -XX:+UnlockExperimentalVMOptions -XX:+UseG1GC -XX:G1NewSizePercent=20 -XX:G1ReservePercent=20 -XX:MaxGCPauseMillis=50 -XX:G1HeapRegionSize=32M
	ModLauncher: 4.1.0+62+5bfa59b
	ModLauncher launch target: fmlclient
	ModLauncher naming: srg
	ModLauncher services: 
		/eventbus-1.0.0-service.jar eventbus PLUGINSERVICE 
		/forge-1.14.4-28.2.16.jar object_holder_definalize PLUGINSERVICE 
		/forge-1.14.4-28.2.16.jar runtime_enum_extender PLUGINSERVICE 
		/accesstransformers-1.0.5-shadowed.jar accesstransformer PLUGINSERVICE 
		/forge-1.14.4-28.2.16.jar capability_inject_definalize PLUGINSERVICE 
		/forge-1.14.4-28.2.16.jar runtimedistcleaner PLUGINSERVICE 
		/OptiFine_1.14.4_HD_U_F5.jar OptiFine TRANSFORMATIONSERVICE 
		/forge-1.14.4-28.2.16.jar fml TRANSFORMATIONSERVICE 
		/MixinBootstrap-1.0.3.jar mixinbootstrap TRANSFORMATIONSERVICE 
		/OptiForge-MC1.14.4-0.1.1.jar optiforge TRANSFORMATIONSERVICE 
		/MixinBootstrap-1.0.3.jar mixin PLUGINSERVICE 
		/MixinBootstrap-1.0.3.jar mixinbootstrap PLUGINSERVICE 
	FML: 28.2
	Forge: net.minecraftforge:28.2.16
	FML Language Providers: 
		javafml@28.2
		minecraft@1
	Mod List: 
		forge-1.14.4-28.2.16-client.jar Minecraft {minecraft@1.14.4 DONE}
		OptiForge-MC1.14.4-0.1.1.jar OptiForge {optiforge@0.1.1 DONE}
		4.jar VoxelMap {voxelmap@1.9.13 DONE}
		2.jar MaLiLib {malilib@0.10.0-dev.19 DONE}
		forge-1.14.4-28.2.16-universal.jar Forge {forge@28.2.16 DONE}
		ImmersiveEngineering-0.14-96.jar Immersive Engineering {immersiveengineering@0.14-96 DONE}
		3.jar RandomPatches {randompatches@1.14.4-1.19.1.1 DONE}
		create-mc1.14.4_v0.2.3.jar Create {create@mc1.14-0.2.3 DONE}
		5.jar Durability101 {durability101@0.0.3 DONE}
		1.jar AppleSkin {appleskin@1.0.13 DONE}
		6.jar Just Enough Items {jei@6.0.0.26 DONE}
	Launched Version: 1.14.4-forge-28.2.16
	LWJGL: 3.2.2 build 10
	OpenGL: Mesa Intel(R) UHD Graphics 620 (WHL GT2) GL version 4.6 (Compatibility Profile) Mesa 20.0.6, Intel
	GL Caps: Using GL 1.3 multitexturing.
Using GL 1.3 texture combiners.
Using framebuffer objects because OpenGL 3.0 is supported and separate blending is supported.
Shaders are available because OpenGL 2.1 is supported.
VBOs are available because OpenGL 1.5 is supported.

	Using VBOs: Yes
	Is Modded: Definitely; Client brand changed to 'forge'
	Type: Client (map_client.txt)
	Resource Packs: 
	Current Language: English(US)
	CPU: 8x Intel(R) Core(TM) i5-8265U CPU @ 1.60GHz
	OptiFine Version: OptiFine_1.14.4_HD_U_F5
	OptiFine Build: 20191204-141250
	Render Distance Chunks: 10
	Mipmaps: 4
	Anisotropic Filtering: 1
	Antialiasing: 0
	Multitexture: false
	Shaders: base
	OpenGlVersion: 4.6 (Compatibility Profile) Mesa 20.0.6
	OpenGlRenderer: Mesa Intel(R) UHD Graphics 620 (WHL GT2)
	OpenGlVendor: Intel
	CpuCount: 8
```

**Comment by facetorched** [comment]
Looks like you are using a shader and it is running into issues with rendering the core sample item. I did the same thing you did and I'm not getting crashes. As soon as I use a shaderpack my game crashes with the same error.
Furthermore, is it ok to compile the code myself and use it as if I got it off of curseforge? I'm too impatient to wait for another update on curseforge.
